eventsourcing/src/main/java/org/axonframework/eventsourcing/EventStateApplier.java

```diff
@@ -14,7 +30,7 @@
  * @since 5.0.0
  */
 @FunctionalInterface
-public interface EventStateApplier<M> extends BiFunction<M, EventMessage<?>, M> {
```

Comment:
What do you think about renaming it to `EventStateEvolver` and the method to `evolve`? 
I left it `apply`, because it's better shows the intention - we'd like to apply event and evolve state into the new one (return it as result), we don't want to "change" / modify the state (of course it may happen, based on the @EventHandler implementation.

Reply:
We need to see what fits best. I think this would become a `EventSourcingMessageHandlerComponent`, under what we discussed last. We need to be sure that that covers this `evolve` idea good enough.
I think we discussed the decider/evolve model, but I'm not sure if we decided to use that terminology, or the one we already had.

Reply:
I agree, let's see it in the future how will it develop.